docs/4.x/vue-getting-started.md

```diff
@@ -0,0 +1,128 @@
+
+Example:
+
+```typescript
+import { reactive, computed } from 'vue';
+
+class MyStore {
+  private myState = reactive({
+    counter: 0,
+  });
+
+  readonly counter = computed(() => this.counter);
+  readonly isZero = computed(() => this.counter === 0);
+
+  increment() {
+    this.counter += 1;
+  }
+  
+  decrement() {
+    this.counter -= 1;
+  }
+}
+
+export default new MyStore();
+```
+
+```vue
+<template>
+  <div>
+    <h2>Counter value is: {{ counter }}</h2>
+    <div v-if="isZero">The counter is at zero! Press increment or decrement to change the value.</div>
+    <div>
+      <button @click="increment()">Increment</button>
+      <button @click="decrement()">Decrement</button>
+    </div>
+  </div>
+</template>
+<script>
+import { defineComponent } from 'vue';
+import MyStore from './MyStore';
+
+export default defineComponent({
+  setup() {
+    return {
+      counter: MyStore.counter,
+      isZero: MyStore.isZero,
+      increment: MyStore.increment.bind(MyStore),
+      decrement: MyStore.decrement.bind(MyStore),
+    };
+  },
+});
+</script>
+```
+
+Classes implementing the store pattern should:
+
+* define private reactive state
+* only provide public readonly access to that state. This readonly access should be deep, no part of the private
+  state should be modifiable by other classes.
+* use computed properties, not methods, to return derived data
+* provide public methods for logic that mutates the state
+
+### Accessing and changing the URL
+
+URLs in Matomo are mainly based on query parameters, the path and host are not normally used. The base URL's
+search has a query string, and the URL hash has another query string. Both are used to determine what the
+query parameter values are, with the hash query parameters overriding the search ones.
+
+In new Vue code, the URL query parameters can be accessed and changed via the MatomoUrl store. This store
+provides a computed parsed `parsed` for easily accessing query parameter values. It also provides a method,
+`updateHash()` that allows developers to change the URL, and thus potentially load a new page.
+
+An example of accessing query parameter values and modifying the hash when needed:
+
+```typescript
+import { computed } from 'vue';
+import { MatomoUrl } from 'CoreHome';
+
+class GoalsStore
+{
+  readonly state = reactive({
+    goals: {}, // maps idGoal => goal. filled out via an ajax request not displayed
+  });
+  
+  readonly currentGoal = computed(() => {
+    const idGoal = MatomoUrl.parsed.idGoal;
+    if (idGoal && this.state.goals[idGoal]) {
+      return this.state.goals[idGoal];
+    }
+    return undefined;
+  });
+  
+  changeGoal(idGoal: number): void {
+    MatomoUrl.updateHash({
+      // NOTE: updateHash will rewrite the entire hash, so it is important to include existing query parameters,
+      // if you only want to overwrite one or a few parameters.
+      ...MatomoUrl.parsed.value,
+      idGoal,
```

Comment:
Haven't tested that though. But would that result in having something like `&idGoal=4` in the url, or only `&4`

Reply:
That is just shorthand for `idGoal: idGoal`, so it would result in merging MatomoUrl.parsed.value with `'idGoal': 4`. So `&idGoal=4`. Btw, there's actually an eslint check that will find `idGoal: idGoal` like code and say it should use the shorthand, so a lot of these things aren't necessary to know beforehand when coding.